api/python/cellxgene_census/src/cellxgene_census/_open.py

```diff
@@ -31,13 +38,15 @@
 api_logger.addHandler(logging.StreamHandler())
 
 
-def _open_soma(locator: CensusLocator, context: Optional[soma.options.SOMATileDBContext] = None) -> soma.Collection:
+def _open_soma(
+    locator: ResolvedCensusLocator, context: Optional[soma.options.SOMATileDBContext] = None
+) -> soma.Collection:
     """Private. Merge config defaults and return open census as a soma Collection/context."""
-    context = _build_soma_tiledb_context(locator.get("s3_region"), context)
+    context = _build_soma_tiledb_context_s3(locator.get("region"), context)
```

Comment:
I suggest you go ahead and scaffold out the provider handling code, even if it boils down to:
```python
if provider in ['s3', 'unknown']:
    ... set region
else:
    ... don't set region
```

Reply:
I implemented this logic:
1. if provider == S3, then build the context using the region
2. otherwise (which includes provider == unknown), just use the passed context.

I think this makes sense because:
- If the user passes a file URI to the `uri` param, we won't need any special context to use it
- If the user passes an s3 uri with a private copy of the census (provider == unknown and uri starts with s3://), they will likely need to specify the region in a custom context. According to the [TileDB docs](https://docs.tiledb.com/main/how-to/backends/s3), a region is mandatory for non us-east-1 buckets, so in theory we could set a specific if for this condition, but it would only work if the custom bucket is in us-east-1. Not worth imho.
- If the user wants to use any other storage medium, they will also need to provide the configuration in the custom context.

Let me know if this looks good.

Reply:
One clarification - we also need to ensure that the default configs are set (e.g., soma.init_buffer_bytes) when the user does *not* specify a context.  So I think the logic is:

if user-provided context, use it.  Else create one with default config.
if provider == S3, set region & anon access flag